src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs

```diff
@@ -95,8 +100,10 @@ protected virtual void InjectDirective()
             var hasTypeError = !At(CSharpSymbolType.Identifier);
             if (hasTypeError)
             {
-                Context.OnError(CurrentLocation,
-                                Resources.FormatMvcRazorCodeParser_KeywordMustBeFollowedByTypeName(InjectKeyword));
+                Context.OnError(
+                    startLocation,
+                    Resources.FormatMvcRazorCodeParser_KeywordMustBeFollowedByTypeName(InjectKeyword),
+                    keywordwithSingleWhitespaceLength + remainingWhitespaceLength);
```

Comment:
:+1: I am removing whitespace from this error. Will also change `@inherits` to not include the trailing whitespace.